src/scene/scene_lua.inl

```diff
@@ -140,63 +141,64 @@ std::vector<openspace::properties::Property*> findMatchesInAllProperties(
         }
     }
 
-    // Stores whether we found at least one matching property. If this is false at the end
-    // of the loop, the property name regex was probably misspelled.
-    for (properties::Property* prop : properties) {
-        // Check the regular expression for all properties
-        const std::string id = prop->uri();
-
-        if (isLiteral && id != propertyName) {
-            continue;
-        }
-        else if (!propertyName.empty()) {
-            size_t propertyPos = id.find(propertyName);
-            if (propertyPos != std::string::npos) {
-                // Check that the propertyName fully matches the property in id
-                if ((propertyPos + propertyName.length() + 1) < id.length()) {
-                    continue;
-                }
-
-                // Match node name
-                if (!nodeName.empty() && id.find(nodeName) == std::string::npos) {
-                    continue;
+    std::mutex mutex;
+    std::for_each(
+        std::execution::par_unseq,
+        properties.cbegin(),
+        properties.cend(),
+        [&](properties::Property* prop) {
+            // Check the regular expression for all properties
+            const std::string_view id = prop->uri();
+
+            if (isLiteral && id != propertyName) {
+                return;
+            }
+            else if (!propertyName.empty()) {
+                size_t propertyPos = id.find(propertyName);
+                if (
+                    (propertyPos == std::string::npos) ||
+                    // Check that the propertyName fully matches the property in id
+                    ((propertyPos + propertyName.length() + 1) < id.length()) ||
+                    // Match node name
+                    (!nodeName.empty() && id.find(nodeName) == std::string::npos))
```

Comment:
Do we want to consider splitting up this if statement using variables? It's a beast (but surprisingly easy to understand thanks to the comments so maybe not an issue. Maybe add a comment to the first line as well?  )

Reply:
The point of turning it into a single statement is to make use of shortcircuiting of the statements.  If the first check already hits, we don't need to do the remaining checks and can save some performance.

Reply:
Yeah, good point 👍